src/plugins/vis_types/pie/public/to_ast.ts

```diff
@@ -65,7 +65,7 @@ export const toExpressionAst: VisToExpressionAst<PartitionVisParams> = async (vi
     nestedLegend: vis.params?.nestedLegend ?? false,
     truncateLegend: vis.params.truncateLegend,
     maxLegendLines: vis.params.maxLegendLines,
-    legendSize: vis.params.legendSize,
+    legendSize: vis.params?.legendSize,
```

Comment:
This `args` object seems a bit problematic. If the `vis.params` can be undefined, then every access to `vis.params` needs to be guarded against that, if instead the `vis.params` is always defined then you can remove completely the optional chainings here.

Reply:
Good point. The `args` object is a patchwork of certainty and uncertainty around the same `vis.params` object.  @stratoula assured me that it's always defined, so I've removed all the optional chaining operators.